### PR TITLE
Handling exception on individual callback

### LINKED ISSUE #28167: [Issue] Handling exception on individual callback
This issue is automatically created based on existing pull request: magento/magento2#27134: Handling exception on individual callback

---------

### Description
This PR changes the behaviour how callbacks are executed after DB changes are committed. The issue was if there is an exception from one call back it results in failing other callbacks as well. 

So put the try/catch on individual callback instead of entire foreach block. 

### Manual testing scenarios (*)
1. Have multiple callbacks having one with fatal error.
2. The subsequenct callbackss after one with fatal error won't get executed.

### PR DESCRIPTION
**Pull request by alok0590**
### Description
This PR changes the behaviour how callbacks are executed after DB changes are committed. The issue was if there is an exception from one call back it results in failing other callbacks as well. 

So put the try/catch on individual callback instead of entire foreach block. 

### Manual testing scenarios (*)
1. Have multiple callbacks having one with fatal error.
2. The subsequenct callbackss after one with fatal error won't get executed. 

### Resolved issues:
1. [x] resolves magento/magento2#28167: Handling exception on individual callback

### CODE REVIEW

**Review by ihor-sviziev — CHANGES_REQUESTED** [review]
Hi @alok0590,
Your changes looks good.
Could you sign Adobe CLA, fix static tests and cover your changes with some tests?

**Review by ihor-sviziev — DISMISSED** [review]

**Comment by magento-engcom-team** [comment]
Hi @ihor-sviziev, thank you for the review.
[ENGCOM-7136](https://jira.corp.magento.com/browse/ENGCOM-7136) has been created to process this Pull Request

**Comment by magento-engcom-team** [comment]
@alok0590 thank you for contributing. Please accept Community Contributors team invitation [here](https://github.com/orgs/magento/invitation) to gain extended permissions for this repository.

**Comment by engcom-Charlie** [comment]
The manual testing scenario covered by a test.

**Comment by engcom-Charlie** [comment]
:heavy_check_mark: QA Passed
For example, we have 4 callbacks and the first callback has a fatal error.
**Before**: The critical exception was logged and the next callbacks don't execute.
**After**: The critical exception was logged and next callbacks execute.

File: lib/internal/Magento/Framework/Model/ResourceModel/AbstractResource.php [hunk 1]
```diff
@@ -92,14 +93,15 @@ public function commit()
          */
```
Comment (slavvka) [inline lib/internal/Magento/Framework/Model/ResourceModel/AbstractResource.php:94]: The logic from this method was moved to `\Magento\Framework\Model\ExecuteCommitCallbacks`. Here it is left only for compatibility. That plugin basically cleans the pool of callbacks here and no callback should be executed. Could you please mark this method as deprecated. 

Also in order for this fix to be complete please fix the same here as well `\Magento\Framework\Model\ExecuteCommitCallbacks`

**Review by slavvka — CHANGES_REQUESTED** [review]

File: lib/internal/Magento/Framework/Model/ResourceModel/AbstractResource.php (hunk 1 above)
Comment (engcom-Charlie) [reply lib/internal/Magento/Framework/Model/ResourceModel/AbstractResource.php:94]: The plugin `\Magento\Framework\Model\ExecuteCommitCallbacks` has already this fix.

Comment (ihor-sviziev) [reply lib/internal/Magento/Framework/Model/ResourceModel/AbstractResource.php:94]: Can confirm that:
https://github.com/magento/magento2/blob/419c4168d4396cac5d1d0a6d3d741eb71861a145/lib/internal/Magento/Framework/Model/ExecuteCommitCallbacks.php#L38-L52

**Review by ihor-sviziev — DISMISSED** [review]

**Comment by magento-engcom-team** [comment]
Hi @ihor-sviziev, thank you for the review.
[ENGCOM-7136](https://jira.corp.magento.com/browse/ENGCOM-7136) has been created to process this Pull Request

**Comment by engcom-Charlie** [comment]
:heavy_check_mark: QA Passed
For example, we have 4 callbacks and the first callback has a fatal error.
**Before**: The critical exception was logged and the next callbacks don't execute.
**After**: The critical exception was logged and next callbacks execute.

**Comment by slavvka** [comment]
@magento create issue

**Review by slavvka — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @slavvka, thank you for the review.
[ENGCOM-7136](https://jira.corp.magento.com/browse/ENGCOM-7136) has been created to process this Pull Request

**Comment by alok0590** [comment]
@slavvka @ihor-sviziev Anything I need to do to get this PR merged ?

**Comment by ihor-sviziev** [comment]
@alok0590 for now no actions required from your side.

**Review by slavvka — COMMENTED** [review]
@alok0590 please resolve the conflicts and re-run tests after upgrade to PHP 7.4

**Review by slavvka — CHANGES_REQUESTED** [review]
@alok0590 please resolve the conflicts and re-run tests after upgrade to PHP 7.4

**Review by ihor-sviziev — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @ihor-sviziev, thank you for the review.
[ENGCOM-7136](https://jira.corp.magento.com/browse/ENGCOM-7136) has been created to process this Pull Request

**Comment by engcom-Charlie** [comment]
:heavy_check_mark: QA Passed
For example, we have 4 callbacks and the first callback has a fatal error.
**Before**: The critical exception was logged and the next callbacks don't execute.
**After**: The critical exception was logged and next callbacks execute.

**Comment by engcom-Charlie** [comment]
Failed functional test not related to changes from this PR.

**Pull request merged**